x-pack/plugins/reporting/server/lib/store/store.ts

```diff
@@ -103,20 +117,21 @@ export class ReportingStore {
   /*
    * Called from addReport, which handles any errors
    */
-  private async indexReport(report: Report) {
-    const doc = {
+  private async indexReport(report: Report): Promise<IndexResponse> {
+    const client = await this.getClient();
+    const { body } = await client.index({
       index: report._index!,
       id: report._id,
+      refresh: true,
```

Comment:
This fixed a bug where a Task Manager could delegate a Kibana instance to run a report, before the reporting document was fully stored in the `.reporting` index